plugins/base/src/main/kotlin/transformers/documentables/DefaultDocumentableMerger.kt

```diff
@@ -41,9 +83,24 @@ private fun <T> mergeExpectActual(
     reducer: (T, T) -> T
 ): List<T> where T : Documentable, T : WithExpectActual {
 
-    fun analyzeExpectActual(sameDriElements: List<T>) = sameDriElements.reduce(reducer)
+    fun analyzeExpectActual(sameDriElements: List<T>) = sameDriElements
+        .partition { it.expectPresentInSet != null }
+        .let {
+            (expects, actuals) -> expects.map { expect ->
+                listOf(expect) + actuals.filter { actual -> DefaultDocumentableMerger.dependencyInfo?.get(actual.sourceSets.single())?.contains(expect.expectPresentInSet!!) ?: false }
```

Comment:
It's use of hack that I commented on top of file. What is going here is just merging to list expect with all actuals that depend on sourceset in which expect is declared.